### PR TITLE
API reference for the Financial Actions API

### PR DESCRIPTION
**Pull request by vasco-grilo-cko**
# Description of changes in PR

[//]: # 'Added the API reference for the upcoming Financial Actions API in NAS.'

## Checklist

- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [x] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

### Comments

All the descriptions were grabbed from https://www.checkout.com/docs/reporting/reconciliation-reports/financial-actions-reports#Report_fields and have been approved by Seb Putman, PM of this product.

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

**Comment by natasha-moore-cko** [comment]
Hi @vasco-grilo-cko, is the PR ready for Tech Writer review?

File: nas_spec/components/schemas/FinancialActions/FinancialAction.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    description: Unique ID of the foreign exchange trade executed in fulfilment of a financial action.
+    example: trd_intsaxljgi6uzkxnv6lex3xayu
+  processed_on:
+    type: string
+    description: The timestamp of when the action was processed and had impacted your balances in ISO format in UTC.
+    format: date-time
+    example: '2022-02-18T13:00:12.357Z'
+  requested_on:
+    type: string
+    description: The timestamp of when a gateway request was made for an action expressed in ISO format in UTC.
```
Comment (natasha-moore-cko) [inline nas_spec/components/schemas/FinancialActions/FinancialAction.yaml:96]:
```suggestion
    description: An ISO 8601 timestamp of when a gateway request was made for an action, in UTC.
```

File: nas_spec/components/schemas/FinancialActions/FinancialActionBreakdown.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    type: number
+    description: The foreign exchange rate applied to a processed financial action.
+    example: 1.24
+  holding_currency:
+    type: string
+    description: Three-digit ISO currency code of the currency a financial action is held and paid out in.
+    example: USD
+  holding_currency_amount:
+    type: number
+    description: The associated amount of the financial action in the held currency.
```
Comment (natasha-moore-cko) [inline nas_spec/components/schemas/FinancialActions/FinancialActionBreakdown.yaml:24]:
```suggestion
    description: The associated amount of the financial action in the holding currency.
```

File: nas_spec/components/schemas/FinancialActions/FinancialActionBreakdown.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    example: 0.19526938
+  processing_currency:
+    type: string
+    description: Three-digit ISO currency code of a financial action when processed.
+    example: GBP
+  processing_currency_amount:
+    type: number
+    description: The associated amount of the financial action in the processed currency.
```
Comment (natasha-moore-cko) [inline nas_spec/components/schemas/FinancialActions/FinancialActionBreakdown.yaml:32]:
```suggestion
    description: The associated amount of the financial action in the processing currency.
```

File: nas_spec/components/schemas/FinancialActions/FinancialActionBreakdown.yaml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    type: string
+    description: The additional level of granularity for predicted scheme fees applied.
+    example: Visa Fixed Acquirer Network Fee
+  reserve_rate:
+    type: string
+    description: The percentage amount reserved from the payment.
+    example: 5%
+  reserve_release_date:
+    type: string
+    description: The date that the reserve is scheduled to be released for a Rolling Reserve Deducted or the date that the reserve is released for a Rolling Reserve Released financial action.
```
Comment (natasha-moore-cko) [inline nas_spec/components/schemas/FinancialActions/FinancialActionBreakdown.yaml:60]:
```suggestion
    description: The date that the reserve is scheduled to be released for a Rolling Reserve Deducted, or the date that the reserve is released for a Rolling Reserve Released financial action.
```

File: nas_spec/components/schemas/FinancialActions/FinancialActionErrorResponse.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+type: object
+properties:
+  request_id:
+    type: string
+    description: Request Id
```
Comment (natasha-moore-cko) [inline nas_spec/components/schemas/FinancialActions/FinancialActionErrorResponse.yaml:5]:
```suggestion
    description: Request ID
```

File: nas_spec/components/schemas/FinancialActions/FinancialActionListResponse.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      type: object
+      $ref: '#/components/schemas/FinancialAction'
+    description: The list of financial actions from the current page.
+  _links:
+    type: object
+    description: The links object.
+    properties:
+      self:
+        type: object
+        description: The self object.
```
Comment (natasha-moore-cko) [inline nas_spec/components/schemas/FinancialActions/FinancialActionListResponse.yaml:28]:
```suggestion
        description: The `self` object.
```

File: nas_spec/components/securitySchemes/OAuth.yaml [hunk 7]
```diff
@@ -74,3 +74,7 @@ flows:
       middleware: Access to all Middleware resources
       'middleware:merchants-secret': Allows merchant identity and Middleware token exchange from a secure execution environment
       'middleware:merchants-public': Allows merchant identity and Middleware token exchange from an insecure execution environment
+      reports: Access to all Reports resources
+      'reports:view': View reports
+      financial-actions: Access to all FinancialActions resources
```
Comment (natasha-moore-cko) [inline nas_spec/components/securitySchemes/OAuth.yaml:79]:
```suggestion
      financial-actions: Access to all Financial actions resources
```

**Review by natasha-moore-cko — CHANGES_REQUESTED** [review]

File: nas_spec/components/schemas/FinancialActions/FinancialAction.yaml [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    type: string
+    description: Four-digit code for retail financial services expressed in ISO 18245 format to classify the types of goods or services provided by you.
+    example: 5311
+  fx_trade_id:
+    type: string
+    description: Unique ID of the foreign exchange trade executed in fulfilment of a financial action.
+    example: trd_intsaxljgi6uzkxnv6lex3xayu
+  processed_on:
+    type: string
+    description: The timestamp of when the action was processed and had impacted your balances in ISO format in UTC.
```
Comment (natasha-moore-cko) [inline nas_spec/components/schemas/FinancialActions/FinancialAction.yaml:91]:
```suggestion
    description: An ISO 8601 timestamp of when the action was processed and impacted your balances, in UTC.
```

**Review by natasha-moore-cko — CHANGES_REQUESTED** [review]

File: nas_spec/paths/financial-actions.yaml [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    - Financial Actions
+  security:
+    - OAuth:
+        - financial-actions
+        - financial-actions:view
+    - ApiSecretKey: [ ]
+  summary: Get financial actions
+  operationId: getFinancialActions
+  description: |
+    Returns the list of financial actions and their details.
```
Comment (natasha-moore-cko) [inline nas_spec/paths/financial-actions.yaml:17]:
```suggestion
  description: |
    <span style="font-size: 14px; font-weight: 500; color: rgb(60, 4, 133); background-color: rgb(237, 222, 255); border-radius: 4px; padding: 2px 6px" >Beta</span><p></p>
    Returns the list of financial actions and their details.
```

Adds the Beta label

File: nas_spec/components/schemas/FinancialActions/FinancialActionListResponse.yaml [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        allOf:
+          - $ref: '#/components/schemas/FinancialActionsSelfLink'
+        example:
+          href: https://api.checkout.com/financial-actions?action_id=act_wsnyzbzmr2huxcekoj7qqhxwuy&payment_id=pay_bpo6m22pnb3epe67i3l6ugtxra&limit=5
```
[the remark this answers is not in the document]
Comment (vasco-grilo-cko) [reply nas_spec/components/schemas/FinancialActions/FinancialActionListResponse.yaml]: thanks for spotting, done [0c82f](https://github.com/checkout/checkout-api-reference/pull/825/commits/0c82fbfbdc71bd55b7f748b518ddedde01fb278f)

File: nas_spec/paths/financial-actions.yaml [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+        default: 10
+        example: 5
+    - name: pagination_token
+      in: query
+      description: A token used for pagination when a response contains results across multiple pages.
+      schema:
+        type: string
+  responses:
+    '200':
+      description: Financial actions retrieved succesfully
```
Comment (natasha-moore-cko) [inline nas_spec/paths/financial-actions.yaml:127]:
```suggestion
      description: Financial actions retrieved successfully
```

File: nas_spec/paths/financial-actions.yaml [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+        additionalProperties: false
+    - name: limit
+      in: query
+      description: The number of results you want to retrieve per page. </br> For example, if the total result count is 50 and you use limit=10, you will need to iterate over 5 pages containing 10 elements each to be able to retrieve all of the reports that match your query.
+      schema:
+        type: integer
+        format: int32
+        minimum: 1
+        maximum: 100
+        default: 10
```
[the remark this answers is not in the document]
Comment (vasco-grilo-cko) [reply nas_spec/paths/financial-actions.yaml]: Yes, good spot @natasha-moore-cko. Done

**Review by natasha-moore-cko — DISMISSED** [review]

**Review by sebastian-garcia-cardona-cko — APPROVED** [review]

**Pull request merged**